core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraRJSParser.kt

```diff
@@ -0,0 +1,756 @@
+                        chunkEndOffset.distance.millimeters.toDouble() /
+                            trackSectionLength.distance.millimeters
+                    ),
+                    chunkSlopes,
+                    chunkCurves,
+                    DirectionalMap(
+                        getGradients(chunkSlopes.get(INCREASING), chunkCurves.get(INCREASING)),
+                        getGradients(chunkSlopes.get(DECREASING), chunkCurves.get(DECREASING))
+                    ),
+                    Offset(chunkLength),
+                    chunkStartOffset,
+                    chunkBlockedGauges
+                )
+            trackSectionChunks.add(chunkIdx)
+        }
+        val trackSectionId = builder.trackSection(rjsTrack.id, trackSectionChunks)
+        for (chunkBoundaryIndex in 0 until chunkBoundariesDetectors.size) {
+            val chunkBoundaryDetectorNames =
+                chunkBoundariesDetectors[chunkBoundaryIndex] ?: continue
+            builder.detector(chunkBoundaryDetectorNames, trackSectionId, chunkBoundaryIndex)
+        }
+    }
+
+    // Parse electrifications
+    for (electrification in rjsInfra.electrifications) {
+        for (electrificationRange in electrification.trackRanges) {
+            val applyElectrificationForChunkBetween =
+                { chunk: TrackChunkDescriptor, chunkLower: Distance, chunkUpper: Distance ->
+                    val previousElectrifications =
+                        chunk.electrificationVoltage.subMap(chunkLower, chunkUpper)
+                    for (previousElectrification in previousElectrifications) {
+                        if (
+                            previousElectrification.value != electrification.voltage &&
+                                previousElectrification.value != ""
+                        ) {
+                            logger.warn(
+                                "Electrification conflict on track-range ${electrificationRange.trackSectionID}" +
+                                    "[${previousElectrification.lower + chunk.offset.distance}, " +
+                                    "${previousElectrification.upper + chunk.offset.distance}]: " +
+                                    "${previousElectrification.value} != ${electrification.voltage}"
+                            )
+                        }
+                    }
+
+                    chunk.electrificationVoltage.put(
+                        chunkLower,
+                        chunkUpper,
+                        electrification.voltage
+                    )
+                }
+            builder.applyFunctionToTrackSectionChunksBetween(
+                electrificationRange.trackSectionID,
+                electrificationRange.begin.meters,
+                electrificationRange.end.meters,
+                applyElectrificationForChunkBetween
+            )
+        }
+    }
+
+    // Parse neutral-sections
+    fun parseNeutralRanges(
```

Comment:
We seem to suddenly switch to use local function half way through `parseRJSInfra()`...